### PR TITLE
Feature/ascii

### PR DESCRIPTION
**Pull request by JJPO96**
Implemented the solution purposed by @dsully for issue #1286

### CODE REVIEW

File: flask/helpers.py [hunk 1]
```diff
@@ -504,6 +504,11 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         a :class:`~datetime.datetime` or timestamp.
         If a file was passed, this overrides its mtime.
     """
+    import warnings
```
Comment (ThiefMaster) [inline flask/helpers.py]: Why not import it at the top level?

File: flask/helpers.py [hunk 2]
```diff
@@ -504,6 +504,11 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
...
+    if type(filename_or_fp) is str:
```
Comment (ThiefMaster) [inline flask/helpers.py]: This is not the proper way of checking whether something is a string. Use `isinstance(.., string_types)` (assuming you want to handle both unicode and byte strings).

File: flask/helpers.py [hunk 3]
```diff
@@ -958,3 +963,16 @@ def total_seconds(td):
     :rtype: int
     """
     return td.days * 60 * 60 * 24 + td.seconds
+
+def is_ascii(string_to_test):
```
Comment (ThiefMaster) [inline flask/helpers.py]: I'd just call the argument `string`

File: flask/helpers.py [hunk 4]
```diff
@@ -958,3 +963,16 @@ def total_seconds(td):
...
+    """Returns if a string object is fully ASCII encoded.
+
+    :param string string_to_test: the string to be evaluated
+
+    :returns: True if string_to_test is fully ASCII encoded, False if not
+    :rtype: boolean
+    """
+    if all(ord(char) < 128 for char in string_to_test):
```
Comment (ThiefMaster) [inline flask/helpers.py]: `return all(ord(char) < 128 for char in string_to_test)`

**Comment by JJPO96** [comment]
@ThiefMaster so is it better to delete this branch and create a new updated one? Or do you have any better suggestion?

**Comment by ThiefMaster** [comment]
No, just rebase/amend and then force-push. That will update this PR without adding even more commits.

**Comment by JJPO96** [comment]
Why did the checks failed?

**Comment by ThiefMaster** [comment]
There's a syntax error. Please test code before committing/pushing it...

**Comment by JJPO96** [comment]
Sorry it was a misunderstood with my colleagues.

**Comment by davidism** [comment]
Closing because #2223 added support for UTF-8 filenames.

**Pull request closed without merging**